random_question/weighted_random.py

```diff
@@ -10,11 +10,26 @@
 
 # TODO: Hess should implement!
 def weighted_sampler(data, weights):
-    """Should return a random item from data array, weighted by the weights array"""
-    raise NotImplementedError("Hess should implement!")
+    random_number=random.random()
+    print('this is the:')
+    print(random_number)
+    Dfunction = arrange_array(weights)
+    print(Dfunction)
+    for x in range(len(Dfunction)):
+            if random_number >= Dfunction[x] and random_number <= Dfunction[x+1] :
+                print('this the data chosen' )
+                print(data[x])
+                return data[x]
+
 
 
 
+def arrange_array(weights):
+    tmp=list(weights)
+    for item in range(len(tmp)):
+        if item != 0:
```

Comment:
1- `range` has a `from` parameter. you could've just skipped the first 0
2- python has an `accumulate` function which does exactly what you're trying to do. obviously,**Always** use the built-in functions over your own code - 
```
from itertools import accumulate

print(list(accumulate([0.5, 0.5, 0.05])))
```

Reply:
an advanced question would be - why do I need, and what happens when I use the `list` constructor?

Reply:
Otherwise we have two pointer to the same list. so when we will change weights_list  each test...
Bad, VERY BAD

Reply:
I meant - why do **I**, in the inlinte example I wrote about `accumulate`, need `list`?

Reply:
i didnt understand the question